pkg/envoy/envoyadminclient.go

```diff
@@ -33,16 +33,22 @@ func NewEnvoyAdminClientForSocket(envoySocketDir string) *EnvoyAdminClient {
 
 func (a *EnvoyAdminClient) transact(query string) error {
 	// Use a custom dialer to use a Unix domain socket for an HTTP connection.
+	var conn net.Conn
+	var err error
 	client := &http.Client{
 		Transport: &http.Transport{
-			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) { return net.Dial("unix", a.unixPath) },
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				conn, err = net.Dial("unix", a.unixPath)
+				return conn, err
+			},
 		},
 	}
 
```

Comment:
Shouldn't we do the `defer conn.Close()` here so that it's also closed in case of error on line 49?

Reply:
No, because at this point `conn` has not been initialized yet. Only after calling Post/Get will `conn` have a value.